src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -123,6 +119,14 @@ protected virtual void SetupRequestDispatcher(IHandlerProvider handlerProvider)
         }
     }
 
+    [JsonRpcMethod("exit")]
+    public Task HandleExitNotificationAsync(CancellationToken _)
+    {
+        var lspServices = _lspServices.Value;
```

Comment:
Should we check if the value has been created? We don't want to spin up, in order to exit.

Reply:
These will be created when this type is constructed and any message is sent to the server, so no need to check here.  We also want them to always be created to ensure any disposal code is run (for example disposing the jsonrpc instance happens here).